src/main/scala/com/bazaarvoice/sswf/model/StepResults.scala

```diff
@@ -0,0 +1,91 @@
+package com.bazaarvoice.sswf.model
+
+sealed abstract class StepResult(message: Option[String]) {
+  def isSuccessful: Boolean
+  def isInProgress: Boolean
+}
+
+case class Success(message: Option[String]) extends StepResult(message) {
+  def this() = this(None)
```

Comment:
Since this is a case class, the alternative constructors are equivalent syntax to `Success.apply()` via the transformation to `Success()`, but if it were a regular class, `new Success()` would not work. I think that the fact that it's a case class means that the compiler generates the corresponding companion object method `object Success { def apply() = new Success() }`, which then invokes the alternative constructor.

So in some sense, you can call the same code from java, but it will look like `Success$.apply()` instead, which no one wants to see. ;)